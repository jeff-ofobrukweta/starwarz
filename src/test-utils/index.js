import { mount } from "enzyme";
import { Provider } from "react-redux";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import logger from "redux-logger";
import thunk from "redux-thunk";
import rootReducer from "../redux/rootReducers";

export const testStore = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, logger))
);

export const reduxify = (Component, props = {}, state = {}) => {
  return function reduxWrap() {
    return (
      <Provider store={testStore}>
          <Component {...props} />
      </Provider>
    );
  };
};


export const makeMountRender = (Component, defaultProps = {}) => {
  return (customProps = {}) => {
    const props = {
      ...defaultProps,
      ...customProps
    };
    return mount(<Component {...props} />);
  };
};


export const snapshotify = reactWrapper => {
  return reactWrapper.html();
};