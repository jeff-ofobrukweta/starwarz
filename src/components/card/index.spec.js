import { makeMountRender, reduxify, snapshotify } from "../../test-utils";
import Card from "../card/index";


describe("Card Component Unit Testing", () => {
  it("Snapshot Card Component", () => {
    const wrapper = makeMountRender(reduxify(Card))()
    expect(snapshotify(wrapper)).toMatchSnapshot();
  });
  it("Check Card Props is Valid", () => {
    const item = {
      count: 6,
      next: null,
      previous: null,
      results: [
        {
          title: "A New Hope",
          created: "2014-12-12T11:26:24.656000Z",
          director: "Irvin Kershner",
          edited: "2014-12-15T13:07:53.386000Z",
          episode_id: 5,
          producer: "Gary Kurtz, Rick McCallum",
          release_date: "1980-05-17",
        }
      ]
    }
    
    const wrapper = makeMountRender(reduxify(Card), item)()
    expect(wrapper.prop("item")).toBe(item);
  });
});