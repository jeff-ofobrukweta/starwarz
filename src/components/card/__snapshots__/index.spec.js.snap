// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Unit Testing Check Card Props is Valid 1`] = `"<div class=\\"characters-section\\"><div class=\\"card-main-inner-container\\"><div class=\\"child-main-contain\\"></div></div></div>"`;

exports[`Card Component Unit Testing Snapshot Card Component 1`] = `"<div class=\\"characters-section\\"><div class=\\"card-main-inner-container\\"><div class=\\"child-main-contain\\"></div></div></div>"`;
